### DISCUSSION TITLE
Proposal: DataType Timestamp remove the precision argument

### DISCUSSION THREAD

**Discussion by sundy-li** [databendlabs/databend#8149] [General]
Timestamp datatype now keep the precision which can be used to output the DateTime with a custom style.

For example:

```
MySQL [(none)]> create table t (a Timestamp(3), b Timestamp(6), c Timestamp(0) );
Query OK, 0 rows affected (0.008 sec)

MySQL [(none)]> insert into t values(now(), now(), now());
Query OK, 0 rows affected (0.020 sec)

MySQL [(none)]> select * from t;
+-------------------------+----------------------------+---------------------+
| a                       | b                          | c                   |
+-------------------------+----------------------------+---------------------+
| 2022-10-12 02:27:18.974 | 2022-10-12 02:27:18.975407 | 2022-10-12 02:27:18 |
+-------------------------+----------------------------+---------------------+
1 row in set (0.026 sec)
```

The storage stores the same value for `now()` which represents instants as the number of microseconds (µs) since 1970-01-01 00:00:00+00.  The `precision` is only for output style, so it's useless to keep it in datatype, we can use other functions like: `to_yyyy_mm_dd` to handle the style.

**Comment by andylokandy** [comment]
Does it mean that the default timestamp format style will always be `YYYY-MM-DD hh:mm::ss.6f`?

**Reply by sundy-li** [reply]
Yes. We can:

1. Add  a setting `format_timestamp_precision`
2. Use `timestamp_format(ts, format)` to control the format style.

I'll fill 2 after #8154  is finished.
